sycl/test-e2e/ESIMD/PerformanceTests/instruction_count.py

```diff
@@ -0,0 +1,59 @@
+import os
+import re
+import sys
+
+
+def main(directory, max_count, target_file=None):
```

Comment:
I don't remember, do we have a use case where we won't pass `target_file`? If no, maybe we can just require it and simplify some of the logic.